plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java

```diff
@@ -200,6 +203,32 @@ public Optional<JdbcTableHandle> getTableHandle(ConnectorSession session, Schema
         }
     }
 
+    /*
+     * Bypass DefaultIdentifierMapping when isCaseInsensitiveNameMatching is false,
+     * because AvaticaDatabaseMetaData always returns true for storesUpperCaseIdentifiers (even though the remote object is not stored as uppercase).
+     * The AvaticaConnection#connectionProperties does not allow changing the identifiers' lexical policy and casing.
```

Comment:
this is entire reason for the complications here. Can you check if a newer Avatica version helps?

Reply:
Another cleaner option is to extend `AvaticaDatabaseMetaData` and use that on our side.

Reply:
> Can you check if a newer Avatica version helps

I will check this.

Reply:
The newer driver version doesnt help. They still maintain the casing properties as internal ones. 

 I also looked at extending AvaticaDatabaseMetaData approach. The Driver's `Meta`  interface doesnt provide good extension points to do this. I think it will end up looking more hacky.

Reply:
See `AvaticaDatabaseMetaData` - https://github.com/apache/calcite-avatica/blob/17e2f484b37ed94ebdb5f486c5dafadc433cf611/core/src/main/java/org/apache/calcite/avatica/AvaticaDatabaseMetaData.java#L156-L158